os/net/lwip/src/api/sockets.c

```diff
@@ -367,29 +365,35 @@ void lwip_socket_thread_cleanup(void)
 	netconn_thread_cleanup();
 }
 
-
-
 /**
  * Map a externally used socket index to the internal socket representation.
  *
  * @param s externally used socket index
+ * @param pid process id used to get the corresponding socketlist
+ *
  * @return struct lwip_sock for the socket or NULL if not found
  */
-struct lwip_sock *get_socket(int s)
+struct lwip_sock *get_socket(int s, pid_t pid)
 {
-	struct lwip_sock *sock;
+	struct lwip_sock *sock = NULL;
+	struct socketlist list;
+	FAR struct tcb_s *tcb;
 
 	s -= LWIP_SOCKET_OFFSET;
 
-	if ((s < 0) || (s >= NUM_SOCKETS)) {
+	if ((s < 0) || (s >= CONFIG_NSOCKET_DESCRIPTORS)) {
 		LWIP_DEBUGF(SOCKETS_DEBUG, ("get_socket(%d): invalid\n", s + LWIP_SOCKET_OFFSET));
 		set_errno(EBADF);
 		return NULL;
 	}
 
-	sock = &sockets[s];
+	tcb = sched_gettcb(pid);
+	DEBUGASSERT(tcb);
 
-	if (!sock->conn) {
+	list = tcb->group->tg_socketlist;
```

Comment:
Pls check for validity of tcb-> group before using it.
DEBUGASSERT(tcb && tcb->group);

Reply:
done